src/graph/heterograph_capi.cc

```diff
@@ -35,7 +36,14 @@ DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroCreateUnitGraphFromCOO")
       formats_vec.push_back(ParseSparseFormat(fmt));
     }
     auto code = SparseFormatsToCode(formats_vec);
-    auto hgptr = CreateFromCOO(nvtypes, num_src, num_dst, row, col, code);
+
+    // setup sorted flags
+    bool row_sorted, col_sorted;
+    std::tie(row_sorted, col_sorted) = COOIsSorted(
+        aten::COOMatrix(num_src, num_dst, row, col));
```

Comment:
Adding this check here could potentially slowdown graph construction. We should expose the sorted flags up to the user-facing API:
```python
g = dgl.graph((src, dst), row_sorted=True, col_sorted=True)
```
Probably also need to expose a utility function for checking sorting status:
```python
row_sorted, col_sorted = dgl.utils.is_sorted_srcdst(src, dst)
```
We could push the later utility function to next PR. When creating the livejournal graph, we manually set both flags to true.

Reply:
@jermainewang Given that previously `COOIsSorted()` was being called at each format conversion, and that it's roughly the cost of copying the graph, would it be fine to make it the default to check?

We could have a flag `check_sorted` that would default to `True`, but would be ignored if `row_sorted` was true. That way power users and internal calls could skip the check:
```
g = dgl.graph((src, dst)) # IsSorted check is invoked
g = dgl.graph((src, dst), check_sorted=False) # IsSorted is not invoked
g = dgl.graph((src, dst), row_sorted=True, col_sorted=True) # IsSorted is not invoked
g = dgl.graph((src, dst), row_sorted=True) # IsSorted is not invoked
```

Reply:
I added the `dgl.utils.is_sorted_srcdst` in #2685

Reply:
> @jermainewang Given that previously `COOIsSorted()` was being called at each format conversion, and that it's roughly the cost of copying the graph, would it be fine to make it the default to check?

It is possible that users only want to perform operations on COO thus not involving any format conversion. For example, in graph classification, once each sample is converted to a graph (likely from a COO format), users call `dgl.batch` immediately before any computation. In this case, adding the sorted check will likely slowdown the graph construction.

Reply:
In 26e7364 I changed the `check_sorted` parameter to default to false.